app/controllers/concerns/hydranorth/files_controller_behavior.rb

```diff
@@ -58,6 +58,16 @@ def edit_form
       end
     end
 
+    def update_metadata
+      if @generic_file[:resource_type].include? Sufia.config.special_types['cstr']
+        self.edit_form_class = Hydranorth::Forms::CstrEditForm
+      elsif @generic_file[:resource_type].include? Sufia.config.special_types['ser']
+        self.edit_form_class = Hydranorth::Forms::SerEditForm
+      elsif @generic_file[:resource_type].include? Sufia.config.special_types['thesis']
+        self.edit_form_class = Hydranorth::Forms::ThesisEditForm
+      end
+      super
+    end
```

Comment:
Can we DRY this up a bit? The presenter and edit_form methods are repeating the same pattern as well. 

I also think there's probably not a lot of value in trying to deal with controller logic in a module, when we're not expecting downstream users to mix this in. We don't have to clean that all up, but we can start down that path.

I'm thinking  delete ```update_metadata```, ```presenter```, and ```edit_form``` (or at least trim ```edit_form``` back to just https://github.com/ualbertalib/HydraNorth/blob/70002cc9f611c6834d062c1076e1eacf860905aa/app/controllers/concerns/hydranorth/files_controller_behavior.rb#L45-L48 if that's not already happening in its ```super```.

Then in ```GenericFilesController``` clean up the repeated checks in all three methods into something like ```cstr_resource?```, ```ser_resource?```, ```dataverse_resource?```, and ```thesis_resource?```.

*Then*, in , you should be able to just do something like this:

```ruby
def presenter_class
    super unless @generic_file
    return Hydranorth::CstrPresenter.new(@generic_file) if cstr_resource?
    # ... etc for every resource type, return Hydranorth::GenericFilePresenter.new(@generic_file) as a last result
end

def edit_form_class
  super unless @generic_file
  return Hydranorth::Forms::CstrEditForm.new(@generic_file) if cstr_resource?
  # etc
end
```

This way we don't have to keep overriding every client method of ```presenter_class``` and ```edit_form_class``` with the same logic. We're just patching the actual methods themselves.

WARNING: This is totally untested on my part so there may be some hidden difficulty I'm not anticipating. But I think it's worth starting down the path of disentangling this code now.